app/routes/account.$accountId.$appId/route.tsx

```diff
@@ -121,6 +121,7 @@ export default function AppIdLayout() {
   const { userRoles } = useOutletContext<AccountIdLoaderData>()
   const actionData = useActionData() as DataStruct<AppIdActionData>
 
+  // handle all notifications at the layout level
   useEffect(() => {
```

Comment:
A change we should do in the future. As this code is duplicated in a few places, we should create a `useActionNotification` hook and reuse it.